### ISSUE TITLE
IdsDatagrid: ScrollRowIntoView() moves the parent element, not the ids-datagrid-row

### ISSUE THREAD

**Issue by GMInf** [infor-design/enterprise-wc#1371]
**Describe the bug**
In some cases when running idsDatagrid.scrollRowIntoView(), the parent element is scrolled, not the ids-datagrid.

**To Reproduce**

1. clone enterprise-wc, npm i
2. update virtual-scroll-infinite-scroll.html and add a text above the datagrid
![image](https://github.com/infor-design/enterprise-wc/assets/53523076/10cadb71-c676-4f79-9cdd-84eca505121b)

3. npm start
4. open virtual-scroll-infinite-scroll.html
5. on the dev console, run document.querySelector("#data-grid-1").scrollRowIntoView(1)
6. error: the datagrid moves up and covers the text

Attaching virtual-scroll-infinite-scroll.html file
[virtual-scroll-infinite-scroll.zip](https://github.com/infor-design/enterprise-wc/files/11982266/virtual-scroll-infinite-scroll.zip)

**Expected behavior**
The idsDatagrid should not move up.

**Version**

- ids-enterprise-wc: beta 12
**Screenshots**

before scrollRowIntoView()
![image](https://github.com/infor-design/enterprise-wc/assets/53523076/07e5b6a1-28c7-4ad4-a027-15c63465a9ee)
after scrollRowIntoView()
![image](https://github.com/infor-design/enterprise-wc/assets/53523076/05378615-ba68-4d02-9b6e-558a293ea497)

**Platform**
Infor Application/Team Name: M3 UX
OS Version: Windows 11
Browser Name: [chrome
Browser Version: 113.0.5672.127 (Official Build) (64-bit)

**Additional context**
We are just replacing the sohodatagrid on our app, we cannot apply all the layout components of ids.
This the closes behavior we could do to replicate the issue.

Adding an option to do `this.rowByIndex(rowIndex)?.scrollIntoView?.({block: 'nearest'});` instead could work, it displays the row at the bottom (the same way as sohoDatagrid) instead of the top but it does not scroll the parent element.

![image](https://github.com/infor-design/enterprise-wc/assets/53523076/924513d8-dcdd-4909-9356-041cde2b50da)

**Comment by jdilag90** [comment]
I think issue happens because example text `1234` is placed in the same `<ids-flex-item grow="1">` as the data-grid, which takes up 100% height of its parent by default.

Some solutions could be
1) Set a height on the ids-data-grid component to a pixel value that fits within parent element
2) Move the example text `1234` into its own ids-flex-item/container, so that ids-data-grid can take up 100% of its parent element height. Example changes below based off `virtual-scroll-infinite-scroll.zip`

![grid-scroll-flex](https://github.com/infor-design/enterprise-wc/assets/98347152/993a0208-f4c5-49ae-8fa5-1c635f007638)

**Comment by tmcconechy** [comment]
@GMInf I agree with @jdilag you should put each item in its own flex item..

```
<ids-layout-flex-layout>
 <ids-layout-flex-item>12345</ids-layout-flex-item>
 <ids-layout-flex-item><ids-data-grid></<ids-data-grid>/ids-layout-flex-item>
</ids-layout-flex-layout>
```

This is the way that ids-layout-flex works. Can you do that to solve this?

**Comment by GMInf** [comment]
Hi @tmcconechy and @jdilag90,

We are just replacing the sohodatagrid on our app, we cannot apply all the layout components of ids-enterprice. This the closes behavior I could do to replicate the issue and to show how ids-row.scrollIntoView() behaves in our application (overlaps other elements).

I have tried 1. Set a height on the ids-data-grid component to a pixel value that fits within parent element
But the result is the same on our app. Instead of the datagrid limiting the row.scrollIntoView() to the parent height, it used the grand parent height.
We cannot do 2., we are just replacing the sohodatagrid on our application. We are not using ids layout. 
Could we setup a meeting for this issue?

**Comment by tmcconechy** [comment]
Brief update is its doing this https://stackoverflow.com/questions/11039885/scrollintoview-causing-the-whole-page-to-move/11041376#11041376

But all of the solutions seem to break virtual scrolling in some way. I can now see the issue however. I dont think we need a call just to reproduce the issue. But wondering if you find a workaround if you can let us know.

**Comment by GMInf** [comment]
Thank you for finding the issue. 
Our current workaround is to check if the idsDatagrid has a scrollbar, if it doesn't we don't run idsDatagrid.scrollRowIntoView(). But looking into your stackblitz example, it could also break when we have a scrollbar on one of its parents.

Have you tried `this.rowByIndex(rowIndex)?.scrollIntoView?.({block: 'nearest'});`? It has a different behavior (show on bottom) but I think it will work on our App. I think it would be the same behavior as sohodatagrid.scrollRowIntoView() which scrolls to the bottom.

**Comment by GMInf** [comment]
Just an update with our workaround, it failed on other environments and scenarios. We may need a fix from the IdsDatagrid.scrollRowIntoView(). 

 Have you tried the parameters below on your stackblitz example? I does not move the parent scroll but it shows the element below

```
  document
    .querySelector('#x')
    .scrollIntoView({ block: 'end', inline: 'nearest' });
```
or
```
  document
    .querySelector('#x')
    .scrollIntoView(false);
```

**Comment by tmcconechy** [comment]
@GMInf we may have another fix we are testing that doesn't request. But we looked at `block: 'nearest'` and we thought maybe the behavior was undesirable.

But i think we can make a passthrough option along with a fix if it doesn't break too much.

**Comment by GMInf** [comment]
Hi @tmcconechy,
I was wrong about the behavior of the sohodatagrid.scrollRowIntoView(), it does not display the row on the bottom part. It displays the row on the center of the datagrid. So I think using `{ block: 'nearest' }` is not a good solution for us. Please cancel our request for the passthrough option and wait for your fix.

**Comment by tmcconechy** [comment]
@GMInf I have a somewhat hacky fix on https://github.com/infor-design/enterprise-wc/pull/new/1371-scrollintoview-bug that works for the given case by setting back the parent scroll position.

But before we PR that wondering

a) what you think
b) if it works on your end and you can test it
c) if it is better/possible to just put the elements in a different parent container than the datagrid in order to solve the issue

Maybe we should have a call or is it possible to see a page with this problem? I'd like to see if there is an alternative or restructuring something on your end to fix. Or maybe a link to the app to test?

Coming back to this: https://github.com/infor-design/enterprise-wc/issues/1371#issuecomment-1625538747

**Comment by GMInf** [comment]
Hi @tmcconechy,

The https://github.com/infor-design/enterprise-wc/pull/new/1371-scrollintoview-bug will not work on our end because it is the great, great... grandparent of the idsDatagrid that scrolls up on some scenarios (similar to https://stackblitz.com/edit/js-rh3ve6?file=index.html,style.css,index.js). 

But we were able to fix the scroll issue on our end by changing css of the great, great... grandparent of the idsDatagrid (our main-controller). That element scrolls up when idsDatagrid.scrollRowIntoView is called. We removed the overflow:hidden and it seemed to fix our issue.

Yes, we could have a call to discuss this and to show you how it behaves on our app before we applied the fix.
